rules/windows/privilege_escalation_persistence_phantom_dll.toml

```diff
@@ -0,0 +1,87 @@
+[metadata]
+creation_date = "2020/01/07"
+maturity = "production"
+updated_date = "2020/01/07"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the loading of a non Microsoft Signed DLL that is either missing on a default Windows install (also known as
+Phantom DLL) or that can be loaded from a different location by a native Windows process to persist or elevate
+privileges via privileged file write vulnerabilities.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Suspicious DLL Loaded for Persistence or Privilege Escalation"
+references = [
+    "https://itm4n.github.io/windows-dll-hijacking-clarified/",
+    "http://remoteawesomethoughts.blogspot.com/2019/05/windows-10-task-schedulerservice.html",
+    "https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html",
+    "https://shellz.club/edgegdi-dll-for-persistence-and-lateral-movement/",
+    "https://windows-internals.com/faxing-your-way-to-system/",
+    "http://waleedassar.blogspot.com/2013/01/wow64logdll.html",
+]
+risk_score = 73
+rule_id = "bfeaf89b-a2a7-48a3-817f-e41829dc61ee"
+severity = "high"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence", "Privilege Escalation"]
+type = "eql"
+
+query = '''
+library where dll.name:
```

Comment:
just a spacing nitpick - 

```
library where dll.name:
  (
  "wlbsctrl.dll",
  "wbemcomn.dll",
  "WptsExtensions.dll",
  "Tsmsisrv.dll",
  "TSVIPSrv.dll",
  "Msfte.dll",
  "wow64log.dll",
  "WindowsCoreDeviceInfo.dll",
  "Ualapi.dll",
  "wlanhlp.dll",
  "phoneinfo.dll",
  "EdgeGdi.dll",
  "cdpsgshims.dll",
  "windowsperformancerecordercontrol.dll",
  "diagtrack_win.dll"
  ) and not
(dll.code_signature.subject_name:"Microsoft Windows" and dll.code_signature.status:"trusted")
```

Reply:
I'm not getting CI errors but in dev tools it looks like there's a verification exception on both the dll code signature fields, i.e `Unknown column [dll.code_signature.subject_name]`

If we try Ext the [keyword] type seems to be an unsupported nested type in hierarchy

Reply:
![image](https://user-images.githubusercontent.com/56412096/106028392-e0a36700-6099-11eb-8915-32179bad3482.png)

![image](https://user-images.githubusercontent.com/56412096/106028336-d3867800-6099-11eb-8856-30972a3743bc.png)

![image](https://user-images.githubusercontent.com/56412096/106028514-0af52480-609a-11eb-9a5a-8f0ded24acd5.png)

Reply:
Is `dll.code_signature.*` supported in 7.11 for library events? @rw-access

Reply:
I'm testing on 7.11 and looks like both `dll.code_signature.subject_name` and `dll.code_signature.status` are giving me a verification exception shown in the first screenshot above

Reply:
ah. can you check the mapping to see if that field is in it?

Reply:
looks like `dll.code_signature` isn't in the mapping

Reply:
https://github.com/elastic/package-storage/blob/production/packages/endpoint/0.17.0/data_stream/library/fields/fields.yml#L153